tests/e2e/pact/services/update-prescriptions.ts

```diff
@@ -0,0 +1,190 @@
+  })
+  checkValue = (38 - runningTotal % 37) % 37
+  checkValue = _PRESC_CHECKDIGIT_VALUES.substring(checkValue, checkValue+1)
+  prescriptionID += checkValue
+  return prescriptionID
+}
+
+function setTestPatientIfProd(bundle: fhir.Bundle) {
+  if (process.env.APIGEE_ENVIRONMENT === "prod") {
+    const patient = getResourcesOfType.getPatient(bundle)
+    const nhsNumberIdentifier = getNhsNumberIdentifier(patient)
+    nhsNumberIdentifier.value = "9990548609"
+    patient.name = [
+      {
+        "use": "usual",
+        "family": "XXTESTPATIENT-TGNP",
+        "given": [
+          "DONOTUSE"
+        ],
+        "prefix": [
+          "MR"
+        ]
+      }
+    ]
+    patient.gender = "male"
+    patient.birthDate = "1932-01-06",
+    patient.address = [
+      {
+        "use": "home",
+        "line": [
+          "1 Trevelyan Square",
+          "Boar Lane",
+          "Leeds",
+          "West Yorkshire"
+        ],
+        "postalCode": "LS1 6AE"
+      }
+    ]
+  }
+}
+
+function signPrescription(processCase: ProcessCase) {
+  const prepareRequest = processCase.prepareRequest
+  const prepareResponse = convertFhirMessageToSignedInfoMessage(prepareRequest)
+  const digestParameter = prepareResponse.parameter.filter(p => p.name === "digest")[0] as fhir.StringParameter
+  const timestampParameter = prepareResponse.parameter.filter(p => p.name === "timestamp")[0] as fhir.StringParameter
+  const digest = Buffer.from(digestParameter.valueString, "base64").toString("utf-8")
+  const digestWithoutNamespace = digest.replace(`<SignedInfo xmlns="http://www.w3.org/2000/09/xmldsig#">`, `<SignedInfo>`)
+  const signature = crypto.sign("sha1", Buffer.from(digest, "utf-8"), {
+    key: fs.readFileSync(privateKeyPath, "utf-8"),
+    padding: crypto.constants.RSA_PKCS1_PADDING
+  }).toString("base64")
+  const certificate = fs.readFileSync(x509CertificatePath, "utf-8")
+  const certificateValue = certificate
+    .replace("-----BEGIN CERTIFICATE-----\n", "")
+    .replace("\n-----END CERTIFICATE-----", "")
+  const xmlDSig = `
+<Signature xmlns="http://www.w3.org/2000/09/xmldsig#">
+  ${digestWithoutNamespace}
+  <SignatureValue>${signature}</SignatureValue>
+  <KeyInfo>
+      <X509Data>
+          <X509Certificate>${certificateValue}</X509Certificate>
+      </X509Data>
+  </KeyInfo>
+</Signature>
+`
+  const bundle = processCase.request
+  const provenance = getResourcesOfType.getProvenances(bundle)[0]
+  provenance.signature[0].when = timestampParameter.valueString
+  provenance.signature[0].data = Buffer.from(xmlDSig, "utf-8").toString("base64")
+
+  try {
+    const signatureVerifier = crypto.createVerify("RSA-SHA1")
+    signatureVerifier.update(digest)
+    const verified = signatureVerifier.verify(certificate, signature, "base64")
+    if (!verified) {
+      throw new Error("Signature failed verification")
+    }
+  }
+  catch {
+    throw new Error("Signature failed verification")
+  }
+}
+
```

Comment:
sign prescriptions after updating with new ids and create a pact from updated, signed prescriptions. signature verification also is run before running smoke tests